lib/dialects/abstract/connection-manager.js

```diff
@@ -24,7 +24,7 @@ class ConnectionManager {
     this.config = config;
     this.dialect = dialect;
     this.versionPromise = null;
-    this.dialectName = this.sequelize.options.dialect;
+    this.dialectName = typeof this.sequelize.options.dialect === 'string' ? this.sequelize.options.dialect : this.sequelize.options.dialect.prototype.name;
```

Comment:
Do we want to get the class name here? If so it's not gonna be in the prototype but in the class itself:

```suggestion
    this.dialectName = typeof this.sequelize.options.dialect === 'string' ? this.sequelize.options.dialect : this.sequelize.options.dialect.name;
```

Reply:
That would be the expected behavior, but for some reason it doesn't work without prototype

Reply:
What do we get when we call `.prototype.name`?

Reply:
I see the other dialects define directly in the prototype (https://github.com/sequelize/sequelize/blob/main/lib/dialects/mysql/index.js#L57). If we wanna enforce it for custom dialects as well (which should be fine, I guess?) we gotta document it somewhere.